<div style="background:linear-gradient(135deg,#7a3d00 0%,#b45309 55%,#d97706 100%);border-radius:18px;padding:34px 30px;color:#fff;font-family:Inter,Segoe UI,sans-serif">
  <div style="font-size:12px;letter-spacing:3px;color:#ffe9c7;font-weight:700;text-transform:uppercase">Chapter 12 · Solutions</div>
  <div style="font-size:36px;font-weight:900;line-height:1.1;margin:10px 0 6px">Practice Challenges, Worked Answers ✅</div>
  <div style="font-size:15px;color:#ffe6cc;max-width:700px;line-height:1.6">Full solutions to the five "Standardization & z-scores" challenges. Try them yourself first, then compare.</div>
  <div style="margin-top:16px;font-size:13px;color:#ffe2bf">Statistics, Data Science and AI: A Visual Handbook · John Fisher · 2026</div>
</div>

## ⚙️ Setup

In [1]:
import numpy as np
import pandas as pd
from scipy import stats
rng = np.random.default_rng(122)
print("Ready.")

Ready.


<div style="background:#fef3e2;border-left:5px solid #d97706;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#0891b2;letter-spacing:1px">CHALLENGE 1 · Z BY HAND</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Turn a value into SDs</div>
<div style="color:#4a5578;margin-top:6px">A test has mean 70 and standard deviation 8. Compute the z-score for a score of 86 and for a score of 62, and say what each one means in words.</div>
</div>

In [2]:
mean, sd = 70, 8
for x in [86, 62]:
    z = (x - mean) / sd
    side = "above" if z > 0 else "below"
    print(f"score {x}: z = ({x} - {mean}) / {sd} = {z:+.1f}  ->  {abs(z):.1f} SD {side} the mean")

score 86: z = (86 - 70) / 8 = +2.0  ->  2.0 SD above the mean
score 62: z = (62 - 70) / 8 = -1.0  ->  1.0 SD below the mean


**Answer:** 86 gives **z = +2.0** (two SDs **above** the mean); 62 gives **z = −1.0** (one SD **below**). The sign tells direction, the magnitude tells distance in standard-deviation units.

<div style="background:#fef3e2;border-left:5px solid #d97706;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#7c3aed;letter-spacing:1px">CHALLENGE 2 · ONE COMMON RULER</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Compare different scales</div>
<div style="color:#4a5578;margin-top:6px">You scored 88 in History (class mean 80, sd 5) and 88 in Physics (class mean 75, sd 13). Same raw score, but on which exam did you perform relatively better?</div>
</div>

In [3]:
for subj, m, s in [("History", 80, 5), ("Physics", 75, 13)]:
    z = (88 - m) / s
    print(f"{subj:8} z = (88 - {m}) / {s} = {z:+.2f}")

History  z = (88 - 80) / 5 = +1.60
Physics  z = (88 - 75) / 13 = +1.00


**Answer:** History gives **z = +1.60**, Physics **z = +1.00**. The identical raw 88 is a stronger result in **History**, because it sits further above its class average in standard-deviation terms. Raw scores from different distributions are not comparable; z-scores are.

<div style="background:#fef3e2;border-left:5px solid #d97706;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#d97706;letter-spacing:1px">CHALLENGE 3 · Z TO PERCENTILE</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Only when normal</div>
<div style="color:#4a5578;margin-top:6px">Heights are approximately normal with mean 170 cm and sd 7 cm. A person is 184 cm. What is their z-score, and roughly what percentile is that? What assumption are you relying on?</div>
</div>

In [4]:
mean, sd, x = 170, 7, 184
z = (x - mean) / sd
pct = stats.norm.cdf(z) * 100
print(f"z = ({x} - {mean}) / {sd} = {z:+.2f}")
print(f"percentile ~ {pct:.1f}th   (via the standard normal curve)")

z = (184 - 170) / 7 = +2.00
percentile ~ 97.7th   (via the standard normal curve)


**Answer:** z = **+2.0**, which maps to about the **97.7th percentile**. This percentile reading relies on heights being **approximately normal**. For skewed or heavy-tailed data the z is still valid, but it would NOT translate to that percentile, the empirical 68-95-99.7 rule only holds for roughly normal data.

<div style="background:#fef3e2;border-left:5px solid #d97706;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#059669;letter-spacing:1px">CHALLENGE 4 · OUTLIER HUNT</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Flag the unusual value</div>
<div style="color:#4a5578;margin-top:6px">For the data below, compute z-scores and flag any value with |z| > 2.5 as a potential outlier.</div>
</div>

In [5]:
data = np.array([48, 50, 51, 49, 52, 50, 47, 53, 51, 95])
z = (data - data.mean()) / data.std()
print("z-scores:", np.round(z, 2))
flagged = data[np.abs(z) > 2.5]
print("flagged |z|>2.5:", flagged)

z-scores: [-0.49 -0.34 -0.27 -0.41 -0.19 -0.34 -0.56 -0.12 -0.27  2.98]
flagged |z|>2.5: [95]


**Answer:** The value **95** has the largest z and is flagged as a potential outlier. Caveat worth remembering: because the mean and SD are not robust, one extreme value inflates the SD and can shrink its own z, so on messier data a median/MAD modified z-score is the sturdier test.

<div style="background:#fef3e2;border-left:5px solid #d97706;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#2563eb;letter-spacing:1px">CHALLENGE 5 · SHAPE CHECK</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Standardize, then inspect</div>
<div style="color:#4a5578;margin-top:6px">Take a right-skewed sample rng.lognormal(mean=1, sigma=0.7, size=3000). Standardize it, confirm the standardized data has mean ~0 and sd ~1, and check whether standardizing changed the skewness.</div>
</div>

In [6]:
raw = rng.lognormal(mean=1, sigma=0.7, size=3000)
z = stats.zscore(raw)
print(f"raw : mean {raw.mean():.2f}, sd {raw.std():.2f}, skew {stats.skew(raw):+.2f}")
print(f"std : mean {z.mean():.2f}, sd {z.std():.2f}, skew {stats.skew(z):+.2f}")

raw : mean 3.44, sd 2.85, skew +4.14
std : mean 0.00, sd 1.00, skew +4.14


**Answer:** After standardizing, the mean is **≈ 0** and the sd is **≈ 1**, but the **skewness is unchanged**, the data is just as right-skewed as before. Standardization is a linear transform: it re-centers and re-scales but never reshapes. To reduce skew you need a nonlinear transform like the log (Chapter 11).

---
<div style="background:#ffffff;border:1px solid #e6e9f2;border-radius:16px;padding:22px 26px;font-family:Inter,sans-serif">
<div style="font-size:19px;font-weight:800;color:#1a2138">🎉 Nicely done!</div>
<div style="color:#4a5578;line-height:1.8;margin-top:8px">You computed z-scores by hand, compared performances on different scales, converted a z to a percentile under normality, hunted an outlier, and confirmed that standardizing changes the center and spread but never the shape. The standard scale is now yours to use.</div>
</div>

<div style="text-align:center;color:#8b94b3;font-size:12px;margin-top:16px">Statistics, Data Science and AI: A Visual Handbook · © 2026 John Fisher</div>